packages/frontend-dynamic-feature-loader/src/loader.ts

```diff
@@ -0,0 +1,196 @@
+   */
+  moduleFederation: Omit<Parameters<typeof init>[0], 'name' | 'remotes'>;
+};
+
+/**
+ * A function providing a loader of frontend features exposed as module federation remotes
+ * from the backend dynamic features service.
+ *
+ * @public
+ */
+export function dynamicFrontendFeaturesLoader(
+  options?: DynamicFrontendFeaturesLoaderOptions,
+): CreateAppFeatureLoader {
+  return {
+    getLoaderName() {
+      return 'dynamic-plugins-loader';
+    },
+
+    async load({ config }) {
+      const dynamicPLuginsConfig = config.getOptionalConfig('dynamicPlugins');
+      if (!dynamicPLuginsConfig) {
+        return {
+          features: [],
+        };
+      }
+
+      function error(message: string, err: unknown) {
+        // eslint-disable-next-line no-console
+        console.error(
+          `${message}: ${
+            err instanceof Error ? err.toString() : JSON.stringify(err)
+          }`,
+        );
+      }
+
+      const appPackageName =
+        config.getOptionalString('app.packageName') ?? 'app';
+      let frontendPluginManifests: {
+        [key: string]: string;
+      };
+      try {
+        const apiClient = new DefaultApiClient({
+          discoveryApi: FrontendHostDiscovery.fromConfig(config),
+          fetchApi: {
+            fetch(input) {
+              return global.fetch(input);
+            },
+          },
+        });
+
+        const response = await apiClient.getManifests({});
+        if (!response.ok) {
+          throw new Error(`${response.status} - ${response.statusText}`);
+        }
+        frontendPluginManifests = await response.json();
+        if (typeof frontendPluginManifests !== 'object') {
+          throw new Error(`Invalid Json content: should be a Json object`);
+        }
+      } catch (err) {
+        error(
+          `Failed fetching module federation configuration of dynamic frontend plugins`,
+          err,
+        );
+        return { features: [] };
+      }
+
+      try {
+        init({
+          ...options?.moduleFederation,
+          name: appPackageName
+            .replaceAll('@', '')
+            .replaceAll('/', '__')
+            .replaceAll('-', '_'),
+          remotes: Object.entries(frontendPluginManifests).map(
+            ([name, manifestLocation]) => ({
+              name: name,
+              entry: manifestLocation,
+            }),
+          ),
+        });
+      } catch (err) {
+        error(`Failed initializing module federation`, err);
+        return { features: [] };
+      }
+
+      const features = (
+        await Promise.all(
+          Object.entries(frontendPluginManifests).map(
+            async ([name, manifestLocation]) => {
+              // eslint-disable-next-line no-console
+              console.info(
+                `Loading dynamic plugin '${name}' from '${manifestLocation}'`,
+              );
+              let manifest: Manifest;
+              try {
+                const response = await fetch(manifestLocation);
```

Comment:
I remember in some of the previous testing as we were implementing support in the CLI the manifests didn't work particularly well in practice. Can't remember fully what the issue was though. I do wonder if it makes sense to rely on the hardcoded remote entry point instead? Especially since it avoids this extra fetch.

Reply:
I'd rather depend on the standard, and most flexible module federation way, which is: use the remote manifest in order to take into account all the options that were set when the module federation remote was built.

Reply:
Still I the ability to configure / customize the module federation asset serving through a plugin gives the ability to use the Javascript directly instead of the manifest if desired.
However using the manifest is still the default because it's more powerful and, combined with the configurability mentioned above, can allows things like the change of the public path to accommodate CDN use-cases.